### ISSUE TITLE
installer segfaults when using a custom https mirror with untrusted cert

### ISSUE THREAD

**Issue by hansinator** [clearlinux/clr-installer#634]
**Describe the bug**
when running clr-install with a custom mirror using a HTTPS url it will segfault when the certificate is not trusted.

**To Reproduce**
clr-installer -c my-mirror-install.yaml

**Expected behavior**
log an error message describing whats wrong (certificate error)

**Environment (please complete the following information):**
Distribution:      Clear Linux OS
Installed version: 32050
clr-installer: 2.3.7

**Additional context**
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x5ad87b]

goroutine 1 [running]:
sync.(*RWMutex).RLock(...)
        /usr/lib/golang/src/sync/rwmutex.go:48
github.com/clearlinux/clr-installer/vendor/github.com/leonelquinteros/gotext.(*Locale).GetDomain(0x0, 0x2, 0x1)
        /builddir/build/BUILD/clr-installer-2.3.7/.gopath/src/github.com/clearlinux/clr-installer/vendor/github.com/leonelquinteros/gotext/locale.go:157 +0x2b
github.com/clearlinux/clr-installer/vendor/github.com/leonelquinteros/gotext.(*Locale).Get(0x0, 0x8803e4, 0x17, 0x0, 0x0, 0x0, 0x5, 0x0)
        /builddir/build/BUILD/clr-installer-2.3.7/.gopath/src/github.com/clearlinux/clr-installer/vendor/github.com/leonelquinteros/gotext/locale.go:173 +0x2f
github.com/clearlinux/clr-installer/swupd.SetHostMirror(0xc000019c80, 0x15, 0x883000, 0x1c, 0x0, 0x0, 0x0)
        /builddir/build/BUILD/clr-installer-2.3.7/.gopath/src/github.com/clearlinux/clr-installer/swupd/swupd.go:492 +0x3fb
main.execute(0x100, 0xc000018f60, 0x17, 0x7ffcff0302fa, 0x21, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /builddir/build/BUILD/clr-installer-2.3.7/.gopath/src/github.com/clearlinux/clr-installer/clr-installer/main.go:384 +0x1ba7
main.main()
        /builddir/build/BUILD/clr-installer-2.3.7/.gopath/src/github.com/clearlinux/clr-installer/clr-installer/main.go:115 +0x124

```

**Comment by mdhorn** [comment]
@hansinator Which Clear Linux OS mirror are you using?

If this related to the forum thread, then you can not use a release installer image to install a custom mix. You need to make your own installer image based on your mix.

**Comment by hansinator** [comment]
@mdhorn This also happens if I make my own installer image or just turn any installation into one that is capable of installing. It happens with any of my local nginx based mirrors and I can turn the segfault behaviour on and off by "clrtrust add"-ing or "clrtrust remove"-ing the certificates necessary for my HTTPs connection.

**Comment by hansinator** [comment]
@mdhorn A quick look at the source shows that it basically crashes at this line:
"			return "", fmt.Errorf(utils.Locale.Get("SSL certificate problem"))"
I am no Go developer so I don't know whats happening, but it looks like the Locale thing segfaults while trying to print an error message about my SSL certificate problem. LOL!

**Comment by hansinator** [comment]
@mdhorn This is only one example, I have seen the installer crash frequently with segfaults during error message printing such that it left me having to guess what the real error is.

**Comment by mdhorn** [comment]
The problem appears to be that the Locale is not initialized before the translation call.
Creating a fix.

**Comment by mdhorn** [comment]
Once a new release is made, this Issue will be updated with the release number and the first release of Clear Linux OS which contains the new release of the installer.

**Comment by mdhorn** [comment]
Fixed in release [2.4.0](https://github.com/clearlinux/clr-installer/releases/tag/2.4.0)

**Comment by mdhorn** [comment]
Released in Clear Linux OS 32380
